### ISSUE TITLE
Date filters not working with List Reports

### ISSUE THREAD

**Issue by oaaoaa** [ra-systems/django-slick-reporting#75]
ListReportView is ignoring date filters.

```
class TestList(ListReportView):
    report_title = "Test List"
    report_model = MyModel
    form_class = FilterClass  # subclassed as FilterClass(SlickReportForm, forms.Form) i.e has start_date & end_date fields
    date_field = "dt"

    columns = [("dt", {"verbose_name": "Date"}),
               ("field1", {"verbose_name": "First Field"}),
               ("field2", {"verbose_name": "Second Field"}),]

```
All the filters except for start_date & end_date are applied on the result set. Reason is that in SlickReportingListViewMixin.get_report_generator we build the report_generator without start_date and end_date:
```
        return self.report_generator_class(
            self.get_report_model(),
            q_filters=q_filters,
            kwargs_filters=kw_filters,
            date_field=self.date_field,
            main_queryset=queryset,
            print_flag=for_print,
            limit_records=self.limit_records,
            columns=self.columns,
            format_row_func=self.format_row,
            container_class=self,
        )

```
Whereas in ReportBaseView.get_report_generator we do pass start_date and end_date while building the report_generator:
```
        return self.report_generator_class(
            self.get_report_model(),
            start_date=self.form.get_start_date(),
            end_date=self.form.get_end_date(),
            q_filters=q_filters,
            kwargs_filters=kw_filters,
            date_field=self.date_field,
            main_queryset=queryset,
            print_flag=for_print,
            limit_records=self.limit_records,
            swap_sign=self.swap_sign,
            columns=self.columns,
            group_by=self.group_by,
            group_by_custom_querysets=self.group_by_custom_querysets,
            group_by_custom_querysets_column_verbose_name=self.group_by_custom_querysets_column_verbose_name,
            time_series_pattern=time_series_pattern,
            time_series_columns=self.time_series_columns,
            time_series_custom_dates=self.time_series_custom_dates,
            crosstab_field=self.crosstab_field,
            crosstab_ids=self.crosstab_ids,
            crosstab_columns=self.crosstab_columns,
            crosstab_compute_remainder=crosstab_compute_remainder,
            format_row_func=self.format_row,
            container_class=self,
            doc_type_plus_list=doc_type_plus_list,
            doc_type_minus_list=doc_type_minus_list,
        )

```

**Comment by RamezIssac** [comment]
Thank you for the hint , ListReprotView is a bit lagging indeed.
I will revise and notify you back

**Comment by RamezIssac** [comment]
Hello @oaaoaa 
Published a new release 1.1.0
This issue should be fixed
